src/dfx-core/src/config/model/canister_id_store.rs

```diff
@@ -16,15 +19,23 @@ pub type CanisterIdString = String;
 pub type NetworkNametoCanisterId = BTreeMap<NetworkName, CanisterIdString>;
 pub type CanisterIds = BTreeMap<CanisterName, NetworkNametoCanisterId>;
 
+// Canister timestamp is saved as a num_bigint::BigInt because of serialization problems with candid::Int
+pub type NetworkNametoCanisterTimestamp = BTreeMap<NetworkName, BigInt>;
```

Comment:
u64 should be sufficient.  I would consider storing this in the json as an ISO 8601 timestamp string.


```suggestion
pub type NetworkNametoCanisterTimestamp = BTreeMap<NetworkName, u64>;
```